alacritty/src/display/mod.rs

```diff
@@ -188,9 +188,17 @@ pub struct Display {
 
 impl Display {
     pub fn new<E>(config: &Config, event_loop: &EventLoop<E>) -> Result<Display, Error> {
-        // Guess DPR based on first monitor.
-        let estimated_dpr =
-            event_loop.available_monitors().next().map(|m| m.scale_factor()).unwrap_or(1.);
+        #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+        let is_x11 = event_loop.is_x11();
+        #[cfg(any(feature = "wayland", target_os = "macos", windows))]
+        let is_x11 = false;
+
+        // Guess DPR based on first monitor, except on Wayland.
+        let estimated_dpr = if cfg!(macos) || cfg!(windows) || is_x11 {
```

Comment:
I'm frankly kinda confused why clippy is reporting `is_x11` as unused (in any configuration), since it's used unconditionally here.

Reply:
It's a fair complaint, `is_x11` is always false, so this branch can never happen.

And since the cfgs are compiletime, it just assumes they're all relevant platform checks.

Reply:
But that's not the issue, it's warning me that `is_x11` is never used, yet it clearly is used in this conditional. It's not warning me that a branch is dead.

```
error: unused variable: `is_x11`
   --> alacritty/src/display/mod.rs:192:13
    |
192 |         let is_x11 = event_loop.is_x11();
    |             ^^^^^^ help: if this is intentional, prefix it with an underscore: `_is_x11`
    |
```

I was thinking perhaps it's because Rust is statically evaluating the condition or something, but this smaller program compiles without a warning:

```rust
#[warn(unused_variables)]

fn main() {
    let x = false;
    if cfg!(foo) || x {}
}
```

I feel like I'm missing something obvious.

Reply:
Because `feature = "wayland"` can be enabled together with the x11 feature. So on Linux by default you'll get both. The compiler is right.

Reply:
Ohhhhhh I see, it's complaining that the first one isn't used, but the second one still is. Thanks you!